src/utils/Switch.ts

```diff
@@ -0,0 +1,40 @@
+export const Switch = <T, S>(expression:T, ...cases:Array<CaseLike<T, S>>):S => {
+    const firstMatchingCase = cases.find(c => c.matches(expression));
+    if(!firstMatchingCase) throw new Error(`No cases matched expression with value ${expression}`);
```

Comment:
Just curious if we have this condition we don't get to the default case right ? Maybe I am not reading this correctly.

Reply:
the default case is one of the cases, it is the case that, if it exists, it always matches. so the only time we would hit this is if we had a switch where none of the cases matched and there was no default case.